app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -6,6 +6,16 @@ class ManageIQ::Providers::Redhat::InfraManager::Vm < ManageIQ::Providers::Infra
 
   supports_not :migrate, :reason => _("Migrate operation is not supported.")
 
+  supports :reconfigure_disks do
+    if storage.blank?
+      unsupported_reason_add(:reconfigure_disks, _('storage is missing'))
+    elsif ext_management_system.blank?
```

Comment:
Fine, then leave it just like this. I actually prefer having all checks inside this block/